### ISSUE TITLE
plugdata LV2 assertion failure in Ardour + Pipewire

### ISSUE THREAD

**Issue by lazzarello** [plugdata-team/plugdata#867]
* plugdata compiled from source, main branch
   *  *(same results with binary release v0.7.0)*
* ardour6 from Ubuntu Studio rev 6.9.0~ds0-1build1
* pipewire 0.3.70 compiled from source

Standard Output
`JUCE Assertion failure in juce_LV2_Client.cpp:1407`

Corresponding source code from JUCE repo (this is probably a JUCE bug)
```C++
1402            const auto boundedBlockLength = hasFeature (features, LV2_BUF_SIZE__boundedBlockLength);
1403
1404            if (! boundedBlockLength)
1405            {
1406               // The host doesn't provide the 'bounded block length' feature
1407                jassertfalse;
1408                return nullptr;
1409            }
```

GUI Dialog from Ardour
> The Plugin "plugdata" could not be loaded
   See the Log window for more details (maybe)

Ardour log window
```
023-05-06T14:42:07 [INFO]: Loading plugin order file /home/lee/.config/ardour6/plugin_metadata/plugin_order
2023-05-06T14:42:07 [INFO]: Loading history from /home/lee/Music/Ardour6 Sessions/plugdata/plugdata.history
2023-05-06T14:42:07 [INFO]: plugdata: no history file "/home/lee/Music/Ardour6 Sessions/plugdata/plugdata.history" for this session.
2023-05-06T14:42:50 [ERROR]: LV2: Failed to instantiate plugin https://github.com/timothyschoen/plugdata
```

**Comment by lazzarello** [comment]
Looked into how LV2 plugins define host features and the LV2_BUF_SIZE__boundedBlockLength feature must be defined. I suspect this is a pipewire issue, as the default "quantum" ( I think that's their fancy term for buffer size) is variable.

Commenting out the assertion error logic and loading plugdata segfaults Ardour when it calls the prepareToWrite() function from JUCE.

I'll move this over to that project. plugdata looks so cool!

**Comment by timothyschoen** [comment]
I remember there being an issue with Ardour 6 in the past, though I think it's actually a problem with Ardour instead of an issue with JUCE or plugdata. IIRC, Ardour is missing support for one of the LV2 extensions that JUCE requires, they were working on a fix, I think it might already be fixed in Ardour 7.4

**Comment by lazzarello** [comment]
well then, thanks for the tip. I'm compiling everything else, might as well double down and compile Ardour  :)  I'll close this out of that works.
